displayChallenges.js

```diff
@@ -0,0 +1,282 @@
+            for (const element of listElements) {
+                element.style.width = "auto";
+                element.style.margin = "auto";
+                element.style.alignContent = "center";
+            }
+        } else {
+            for (const element of listElements) {
+                element.style.width = "calc(95%/3)";
+                element.style.maxWidth = "400px";
+                element.style.margin = "0";
+                challengesHtmlList.lastChild.style.marginRight = "auto";
+                challengesHtmlList.lastChild.style.paddingLeft = "2.5%";
+            }
+        }
+    }
+}
+
+
+// Buttons for online challenges
+
+const onlineButtons = document.querySelectorAll("a.button.primary");
+onlineButtons.forEach(button => {
+    button.addEventListener("click", (ev) => {
+        document.cookie = "online";
+        document.location.href = "http://127.0.0.1:5500/challenges.html";
+    })
+})
+
+
+// Buttons for on-site challenges
+
+const onSiteButtons = document.querySelectorAll("a.button.secondary");
+onSiteButtons.forEach(button => {
+    button.addEventListener("click", (ev) => {
+        document.cookie = "offline";
+        document.location.href = "http://127.0.0.1:5500/challenges.html";
+    })
+})
+
+// THE function that interwines everything
+
+function wellHello() {
+    if (document.cookie == "online") {
+        console.log("jag visar online");
+        displayOnlineRooms();
+        document.querySelector("input#online").checked = true;
+    } else if (document.cookie == "offline") {
+        console.log("jag visar offline");
+        displayOnSiteRooms();
+        document.querySelector("input#onsite").checked = true;
+    } else {
+        console.log("jag visar alla");
+        displayAllRooms();
+        document.querySelector("input#online").checked = true;
+        document.querySelector("input#onsite").checked = true;
+    }
+    document.cookie = "";
+
+
+    // TO BE CONTINUED...
+
+
+    //    const onlineCheckBox = document.querySelector("input#online");
+    //   onlineCheckBox.addEventListener("click", (ev) => {
+    //     if (onlineCheckBox.checked == true) {
+    //        displayOnlineRooms();
+    //   } else {
+    //      displayOnSiteRooms();
+    //  }
+    // })
+
+}
+
+
+// Creates a section that is containing a list that WILL display the challenges (depending on what challenges). 
+
+const main = document.querySelector("main");
+const roomSection = document.createElement("section");
+const challengesHtmlList = document.createElement("ul");
+challengesHtmlList.style.display = "flex";
+challengesHtmlList.style.flexWrap = "wrap";
+challengesHtmlList.style.justifyContent = "space-between";
+challengesHtmlList.style.alignContent = "center";
+challengesHtmlList.style.listStyle = "none";
+
+
+// Shows all the challenge rooms (both online/onsite) 
+
+async function displayAllRooms() {
+    const res = await fetch('https://lernia-sjj-assignments.vercel.app/api/challenges');
```

Comment:
It may be simpler to create a API function to call, so we don't have so mutch code  repetition. If you check the main branch I have created an API class in src/script/myAPI.js that you can call, it returns the list.

Reply:
Code repetition is one issue, but the bigger issue is that the same data is retrieved (over the network) over and over again, even if the code is put in a function. A better way would be to _cache_ the data, which is a fancy way of saying to store it in a variable after retrieving it, so that you can reuse it from the variable instead of retrieving it again.